pkg/storage/replica_proposal.go

```diff
@@ -582,6 +579,18 @@ func (r *Replica) handleReplicatedEvalResult(
 		rResult.ComputeChecksum = nil
 	}
 
+	if rResult.RaftLogDelta != nil {
+		r.mu.Lock()
+		r.mu.raftLogSize += *rResult.RaftLogDelta
+		if r.mu.raftLogSize < 0 {
+			// Ensure raftLogSize is not negative since it isn't persisted between
```

Comment:
I don't understand this comment. What does it not being persisted have to do with it going negative?